src/test/java/org/jpeek/graph/DisjointTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2019 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.jpeek.graph;
+
+import java.util.Set;
+import org.cactoos.list.ListOf;
+import org.hamcrest.core.AllOf;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.llorllale.cactoos.matchers.Assertion;
+import org.llorllale.cactoos.matchers.HasValuesMatching;
+
+/**
+ * Test case for {@link Disjoint}.
+ * @since 0.30.9
+ */
+public final class DisjointTest {
+
+    @Test
+    @Disabled
+    @SuppressWarnings("unchecked")
+    public void calculatesDisjointSets() throws Exception {
+        final Node one = new Node.Simple("one");
+        final Node two = new Node.Simple("two");
+        final Node three = new Node.Simple("three");
+        final Node four = new Node.Simple("four");
+        final Node five = new Node.Simple("five");
+        final Node six = new Node.Simple("six");
+        one.connections().add(two);
+        two.connections().addAll(new ListOf<>(one, three));
+        three.connections().add(two);
+        four.connections().add(five);
+        five.connections().add(four);
+        final Graph graph = new FakeGraph(
+            new ListOf<>(one, two, three, four, five, six)
+        );
+        //@checkstyle MagicNumberCheck (50 lines)
+        new Assertion<>(
+            "Must build disjoint sets correctly",
+            new Disjoint(graph).value(),
+            new AllOf<Iterable<Set<Node>>>(
+                new ListOf<>(
+                    new HasValuesMatching<>(
```

Comment:
@HDouss I think that is better we have one matcher for each operation, something like this:
```
set,
new AllOf(
    new IsIterableContaining(
        new IsEqual(one)
    ),
    new IsIterableContaining(
        new IsEqual(two)
    ),
    new IsIterableContaining(
        new IsEqual(three)
    ),
   new IsIterableWithSize(
      new IsEqual(3)
   )
)
```

Reply:
@paulodamaso The problem here, is that we cannot know what if the set (1-2-3) is the first one in the list, (4-5) is the second, and (6) is the third. That will make a big constraint on the future implementation. So the idea is: the list contains a set (1-2-3) without knowing wich set exactly in the list.

Reply:
@paulodamaso what do you think?

Reply:
@HDouss I don't see the problem, we would be using set the same way we are currently using, but we would just change 
```
                    new HasValuesMatching<>(
                        set -> {
                            return set.contains(five) && set.contains(four) && set.size() == 2;
                        }
                    )
```
to avoid the imperative code.

Reply:
@paulodamaso maybe I am misunderstanding something. Your suggested approach takes a well defined set (which is an element of the list) and applies matchers on that set. In my approach, the `AllOf` is applied on the list. The affirmation succeeds if one of the elements of the list (so, any set without knowing which one in advance) returns true giving the function. The matchers are applied to the list, and not to a particular item of the list. Applying matchers on a particular item on the list implies the problem I stated in my previous comment.

Reply:
@paulodamaso Please note that `HasValuesMatching` is a matcher for the list, and not the set.

Reply:
@HDouss Right, now I understand, thanks for the explanation. Just remove `return` statement and the curly braces, we don't need them because we have just one statement on each lambda function.